internal/typeinfo/slice_test.go

```diff
@@ -0,0 +1,45 @@
+package typeinfo
+
+import (
+	"reflect"
+
+	. "gopkg.in/check.v1"
+)
+
+func (*typeInfoSuite) TestLocateParamsSlice(c *C) {
+	type S []any
+
+	argInfo, err := GenerateArgInfo([]any{S{}})
+	c.Assert(err, IsNil)
+
+	s := S{1, 2}
```

Comment:
I think it would not take a lot of refactoring to make this a test array. That way we can add tests easily in the future and we can test boundary conditions like empty slice, mixed types, etc.

Reply:
I've changed it to a table driven test here, there are only two cases right now though.

Reply:
Thanks, right now there are only two tests but in the future we can add more easily :)

Reply:
Tests with arrays of more than one type would be nice if that can be included in this PR.

Reply:
Added that, `S` is a list of any so its not too much of a special case.